src/reference-manual/pedantic-mode.Rmd

```diff
@@ -0,0 +1,166 @@
+Pedantic mode generates a warning when any branching condition may depend on a parameter value.
+
+
+## Parameters with multiple twiddles
+
+A warning is generated when a parameter is found on the left-hand side of more than one ~ statements (or an equivalent `target +=` conditional density statement).
+This pattern is not inherently an issue, but it is unusual and may indicate a mistake.
+
+Pedantic mode only searches for repeated statements, it will not for example generate a warning when a ~ statement is executed repeatedly inside of a loop.
+
+
+## Parameters with zero or multiple priors
+
+A warning is generated when a parameter appears to have greater than or less than one prior distribution factor.
+
+This analysis depends on a [*factor graph*](https://en.wikipedia.org/wiki/Factor_graph) representation of a Stan program. A factor F that depends on a parameter P is called a *prior factor for P* if there is no path in the factor graph from F to any data variable except through P.
+
+
+## Variables used before assignment
+
+A warning is generated when any variable is used before it has been assigned a value.
+This warning is also available as a standalone option to Stanc3, with the flag: `--warn-uninitialized`.
+
+
+## Strict or nonsensical parameter bounds
+
+Parameters that are have strict `upper` and `lower` bounds can cause unmanageably large gradients in a density function, and may only be justified in a few cased.
+A warning is generated for all parameters declared with the bounds `<lower=.., upper=..>` except for `<lower=0, upper=1>` or `<lower=-1, upper=1>`.
+
+In addition, a warning is generated when a parameter bound is found to have `upper - lower <= 0`.
+
+
+## Pedantic Mode limitations
+
+
+#### Constant values are sometimes uncomputable
+
+Pedantic mode attempts to evaluate expressions down to literal values so that they can be used to generate warnings.
+For example, in the code `normal(x, 1 - 2)`, the expression `1 - 2` will be evaluated to `-1`, which is not a valid variance argument so a warning is generated.
+However, this strategy is limited; it is often impossible to fully evaluate expressions in finite time.
+
+
+#### Container types
+
+Currently, indexed variables are not handled intelligently, so they are treated as monolithic variables.
+Each analysis treats indexed variables conservatively (erring toward generating fewer warnings).
+
+
+#### Data variables
+
+The declaration information for `data` variables is currently not considered, so using `data` as incompatible arguments to distributions may not generate the appropriate warnings.
+
+
+#### Control flow dependent on parameters in nested functions
+
+If a parameter is passed as an argument to a user-defined function within another user-defined function, and then some control flow depends on that argument, the appropriate warning will not be thrown.
```

Comment:
I'm for warning about these things as much as possible.

Reply:
Ditto, I just left it for a future improvement because the solution is a bit involved.